ethcore/src/machine.rs

```diff
@@ -132,7 +132,7 @@ impl EthereumMachine {
 	) -> Result<Vec<u8>, Error> {
 		let env_info = {
 			let mut env_info = block.env_info();
-			env_info.gas_limit = env_info.gas_used + gas;
+			env_info.gas_limit = env_info.gas_used.saturating_add(gas);
```

Comment:
It would be great to document (and/or add a test for it) what happens in case of an overflow with this change. It fixes the panic, but is the new non-panicky behaviour legal?

Reply:
I added more docs on how execute_as_system function actually works. I think this is mostly a definition issue -- system transactions are not normal transactions after all.

Reply:
@sorpaas talking further about this with @ascjones we were wondering what the reason is this method accepts a `gas` param at all: as the system calls are "special" and can use however much gas they want, then perhaps this line could just become `env_info.gas_limit = U256::max_value()`?

cc @andresilva

Reply:
It do make sense. Sometimes we want to limit the gas can be used by system transactions. Like BLOCKHASH contract (https://eips.ethereum.org/EIPS/eip-210).

Reply:
@dvdplm It's typically used for consensus-level transactions, e.g. finalizing changes to `ValidatorSet`, block reward contract transactions. In these cases we don't really need to cap the execution using gas since we assume the contracts are well behaved. Although we do provide `U256::max_value()` in most cases, for EIP210 we set the gas limit defined in the spec. So I think we should keep the `gas` parameter and don't assume it will always be `U256::max_value()`.